sql/analyzer/indexed_joins.go

```diff
@@ -292,20 +292,31 @@ func addLookupJoins(m *Memo) error {
 // Ref sction 2.1.1 of:
 // https://www.researchgate.net/publication/221311318_Cost-Based_Query_Transformation_in_Oracle
 // TODO: need more elegant way to extend the number of groups, interner
-func convertSemiToInnerJoin(m *Memo) error {
+func convertSemiToInnerJoin(a *Analyzer, m *Memo) error {
 	seen := make(map[GroupId]struct{})
 	return dfsExprGroup(m.root, m, seen, func(e relExpr) error {
 		semi, ok := e.(*semiJoin)
 		if !ok {
 			return nil
 		}
 
+		rightOutTables := m.tableProps.getTableNames(semi.right.relProps.OutputTables())
+		projectExpressions := []sql.Expression{}
 		onlyEquality := true
 		for _, f := range semi.filter {
 			transform.InspectExpr(f, func(e sql.Expression) bool {
 				switch e.(type) {
-				case *expression.GetField, *expression.Literal, *expression.BindVar,
+				case *expression.GetField:
+					// getField expressions tell us which columns are used, so we can create the correct project
+					gf, _ := e.(*expression.GetField)
```

Comment:
nit: `switch e := e.(type)` makes the cast implicit

Reply:
Good point, changing.